src/v/cloud_storage/partition_manifest.h

```diff
@@ -173,18 +197,38 @@ class partition_manifest final : public base_manifest {
     /// \return true on success, false on failure (no such segment)
     bool delete_permanently(const key& name);
 
+    /// Removes all replaced segments from the manifest.
+    /// Method 'replaced_segments' will return empty value
+    /// after the call.
+    void delete_replaced_segments();
+
     manifest_type get_manifest_type() const override {
         return manifest_type::partition;
     };
 
+    struct replaced_segment {
+        segment_name name;
+        segment_meta meta;
+    };
+
+    /// Return iterators that can be used to access
+    /// segments that were replaced by newer segments.
```

Comment:
maybe this comment should be updated, as the method no longer returns a pair of iterators?